### PR TITLE
fixes: unused variable cleanup

### LINKED ISSUE #548: lint: fix staccheck
* Fix `staticcheck` warnings from the [lint checker](https://github.com/ovrclk/akash/blob/master/.github/workflows/tests.yml#L39) step for it.
 * Remove `continue-on-error: true` from `lint staticcheck` step.

### PR DESCRIPTION
**Pull request by Ropes**
ci-lint detected unused variables returned from function calls and type
comparison. All minor changes.

fixes #548

### CODE REVIEW

File: sdl/units.go [hunk 1]
```diff
@@ -51,8 +52,8 @@ func (u *cpuQuantity) UnmarshalYAML(unmarshal func(interface{}) error) error {
 		if err != nil {
 			return err
 		}
-		if val < 0 {
-			return errNegativeValue
+		if val == 0 {
```
Comment (boz) [inline sdl/units.go:55]: I think `0` is okay here.

File: x/deployment/query/path.go [hunk 2]
```diff
@@ -90,6 +90,9 @@ func ParseGroupPath(parts []string) (types.GroupID, error) {
 	}
 
 	gseq, err := strconv.ParseUint(parts[2], 10, 32)
+	if err != nil {
```
Comment (boz) [inline x/deployment/query/path.go:93]: damn how did these get missed.  thanks!

File: sdl/units.go (hunk 1 above)
Comment (Ropes) [reply sdl/units.go]: fixed

File: sdl/units.go [hunk 3]
```diff
@@ -51,9 +51,6 @@ func (u *cpuQuantity) UnmarshalYAML(unmarshal func(interface{}) error) error {
 		if err != nil {
 			return err
 		}
-		if val < 0 {
-			return errNegativeValue
```
Comment (boz) [inline sdl/units.go:55]: `errNegativeValue` can probably be removed.

Comment (Ropes) [reply sdl/units.go:55]: It's still used below when parsing a float, I think it's fine to leave for now.

**Pull request merged**